### DISCUSSION TITLE
Vtk export question

### DISCUSSION THREAD

**Discussion by laat0003** [modflowpy/flopy#1280] [Q&A] [answered]
Hi,

Hope this is the correct place to ask a question. Will vtk export always be limited to structured grid only? The example notebook provided doesn't appear to state anywhere that you can't use this feature with other grid types. I was trying to use it to export a vtk for a MF6 disu modelgrid but it kept failing with; 'UnstructuredGrid' object has no attribute 'nrow'. I later checked the code for the vtk class and saw an assertion for stuctured grid with a comment saying that vtk only works with structured grids. Although it seems that you never quite reach that assertion unless you do have a structured grid. 

The vtk class has a parameter vtk_grid_type that may take on the value "UnstructuredGrid". This made me think initially that it should work.

Cheers for your time.

**Comment by christianlangevin** [comment]
Hi @laat0003, VTK export should be set up now to work with unstructured grids as long as the vertices and incidence list are provided.  @jlarsen-usgs did a nice rewrite of the VTK export functionality, and if I recall, it is only on the develop branch at the moment through commit https://github.com/modflowpy/flopy/pull/1249.

**Reply by laat0003** [reply]
Hi @langevin-usgs,

Awesome! Thanks very much for pointing that out. I usually use J. Doherty's utilities for vtk export but his MF6 versions are limited to DISV and DIS only so was counting on being able to do this with Flopy. You guys rock.

**Comment by jlarsen-usgs** [comment, marked as answer]
Hi @laat0003, VTK export is set up now to work with all supported modflow 6 grid types in the development version of flopy. If you update flopy using

```
pip install -U https://github.com/modflowpy/flopy/zipball/develop
```

you will have access to the new functionality. The new functionality is described in the example notebook, [vtk_example_notebook](https://github.com/modflowpy/flopy/blob/develop/examples/Notebooks/flopy3_vtk_export.ipynb), and all of the methods to export VTK are the same regardless of grid type. Let me know if you have any other questions!

**Reply by jlarsen-usgs** [reply]
You will need to install the python vtk library to use the new `Vtk` class. 

```
pip install vtk
```

**Reply by laat0003** [reply]
Sweet!

Many thanks for the guidance and your efforts in making this feature available @jlarsen-usgs. Very much appreciate the work you guys put in and also glad I no longer have to hop between Flopy and alternative utilities to get these outputs.  Hope they pay you guys well because this stuff is gold! :)
